build/gulp/tasks/test-projects/webpack/tsconfig.json

```diff
@@ -0,0 +1,10 @@
+{
+  "compilerOptions": {
+    "jsx": "react",
+    "lib": ["dom", "dom.iterable", "esnext"],
+    "module": "esnext",
+    "moduleResolution": "node",
```

Comment:
this is the default value for this option (`node`), so we might skip this from being explicitly specified.

Actually, here I would rather expect that we will test client scenario that does as many overrides as possible for the TS compiler - i.e. the config that is produced by `tsc --init`, - as this mimics default configuration provided by TS team. 

Otherwise, frankly, it is hard for me to understand semantics of this test and, essentially, understand _what_ it verifies - unless we would explicitly state in the docs that this is the _recommended_ TS configuration. However, this would mean that we are imposing additional restriction s to the client (not saying that we shouldn't do that, but we need to justify them first then).

What do you think?

Reply:
> this is the default value for this option (node), so we might skip this from being explicitly specified.

Nope, `module === "AMD" or "System" or "ES6" ? "Classic" : "Node"`

![image](https://user-images.githubusercontent.com/14183168/55715260-0ced7f80-59fd-11e9-954f-af61808d8f86.png)

Reply:
> Actually, here I would rather expect that we will test client scenario that does as many overrides as possible for the TS compiler - i.e. the config that is produced by tsc --init, - as this mimics default configuration provided by TS team.

1. By `tsc --init` there is `strict: true` which will not work with Stardust
2. Without `lib` section nothing will work :(

Reply:
Actually, this config is closer too official tutorial: https://www.typescriptlang.org/docs/handbook/react-&-webpack.html

Reply:
We can do:
```diff
-    "module": "esnext",
-    "moduleResolution": "node",
+    "module": "commonjs",
```

Reply:
After discussion we decided to use plain `tsc`:

![image](https://user-images.githubusercontent.com/14183168/55720801-54c6d380-5a0a-11e9-965e-a89061bcf682.png)